src/NServiceBus.Core/Recoverability/SecondLevelRetries/Config/SecondLevelRetriesSettings.cs

```diff
@@ -26,20 +26,38 @@ public void CustomRetryPolicy(Func<TransportMessage, TimeSpan> customPolicy)
         }
 
         /// <summary>
-        /// Register a custom retry policy.
+        /// Register a custom retry policy. Overrides <see cref="NumberOfRetries"/> and <see cref="TimeIncrease"/> configuration.
         /// </summary>
-        public void CustomRetryPolicy(Func<IncomingMessage, TimeSpan> customPolicy)
+        public SecondLevelRetriesSettings CustomRetryPolicy(Func<IncomingMessage, TimeSpan> customPolicy)
         {
             Guard.AgainstNull(nameof(customPolicy), customPolicy);
-            config.Settings.Set("SecondLevelRetries.RetryPolicy", customPolicy);
+            config.Settings.Set(Recoverability.SlrCustomPolicy, customPolicy);
+
+            return this;
         }
 
         /// <summary>
-        /// Disables second level retries.
+        /// Configures the amount of times a message should be retried with a delay after failing all first level retries.
```

Comment:
Wording is a bit awkward here, at the very least, I'd consider replacing "amount" with "number".